### PR TITLE
Ubuntu 20.04 Support

### LINKED ISSUE #1109: LibreTime support for Ubuntu 20.04 LTS (Focal Fossa)
At the time of writing the LibreTime's installer supports Ubuntu 18.04 (Bionic Beaver) but not Ubuntu 20.04 LTS (Focal Fossa).

About Ubuntu 20.04 LTS (Focal Fossa): this is the latest long-term supported Ubuntu distribution and it will have official support for 5 years until April 2025.

This issue is about landing to this new distribution (where _new_ means 7 months ago at the time of writing).

## What to do

1. [ ] Try Propel v.2 and fix migration issues
2. [ ] package Liquidsoap in Ubuntu (bonus point!)

The issues with 20.04 are listed here: #1024.

## Notes

Another reason is that Liquidsoap and its plugins are not in Ubuntu's official repositories. This is maybe caused by lack of support of these packages since Debian GNU/Linux 10 (buster). The Liquidsoap's plugins were dropped since Debian GNU/Linux 9 (stretch):

* https://packages.debian.org/jessie/liquidsoap-plugin-gstreamer (Debian 8: OK)
* https://packages.debian.org/stretch/liquidsoap-plugin-gstreamer (Debian 9: OK)
* https://packages.debian.org/buster/liquidsoap-plugin-gstreamer (Debian 10: missing)

Anyway I think we can overcome this lack and we may be able to land on Ubuntu 20.20 LTS (Focal Fossa) before 2025.

### PR DESCRIPTION
**Pull request by paddatrapper**
Replaces #1120. Includes better error reporting and correct dependencies. Fixes #1109.

To do:

- [x] Add https://launchpad.net/%7Eondrej/+archive/ubuntu/php/+index?batch=75&memo=150&start=150 PPA for PHP 7.3
- [x] Test

### CODE REVIEW

**Comment by stipic** [comment]
When will be this merged in release?

**Comment by Robbt** [comment]
I tried to install this and it ran into a bunch of issues with various packages.

Also noticed it didn't add the php7.3 backport when the script ran and that the script still mentions php 7.4 in some places whereas we are trying to get php7.3 installed.

Here's what my log showed when I first ran it. The first crop of problems seem related to not running `apt-get update` prior to the script going and the second seem to be an issue with the repository backport of 7.3 not being added.

   ubuntu-focal: Running: /tmp/vagrant-shell20210215-7360-f42ky4.sh
    ubuntu-focal: Reading package lists...
    ubuntu-focal: Building dependency tree...
    ubuntu-focal: Reading state information...
    ubuntu-focal: The following additional packages will be installed:
    ubuntu-focal:   libatopology2 libfftw3-single3 libgomp1 libsamplerate0
    ubuntu-focal: Suggested packages:
    ubuntu-focal:   dialog libfftw3-bin libfftw3-dev
    ubuntu-focal: The following NEW packages will be installed:
    ubuntu-focal:   alsa-utils libatopology2 libfftw3-single3 libgomp1 libsamplerate0
    ubuntu-focal: 0 upgraded, 5 newly installed, 0 to remove and 0 not upgraded.
    ubuntu-focal: Need to get 2,865 kB of archives.
    ubuntu-focal: After this operation, 6,726 kB of additional disk space will be used.
    ubuntu-focal: Err:1 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libatopology2 amd64 1.2.2-2.1ubuntu2.2
    ubuntu-focal:   404  Not Found [IP: 91.189.88.142 80]
    ubuntu-focal: Get:2 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libgomp1 amd64 10.2.0-5ubuntu1~20.04 [102 kB]
    ubuntu-focal: Get:3 http://archive.ubuntu.com/ubuntu focal/main amd64 libfftw3-single3 amd64 3.3.8-2ubuntu1 [756 kB]
    ubuntu-focal: Get:4 http://archive.ubuntu.com/ubuntu focal/main amd64 libsamplerate0 amd64 0.1.9-2 [939 kB]
    ubuntu-focal: Get:5 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 alsa-utils amd64 1.2.2-1ubuntu2 [1,021 kB]
    ubuntu-focal: Fetched 2,818 kB in 1s (2,450 kB/s)
    ubuntu-focal: Unable to correct missing packages.
    ubuntu-focal: W
    ubuntu-focal: : 
    ubuntu-focal: --force-yes is deprecated, use one of the options starting with --allow instead.
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/a/alsa-lib/libatopology2_1.2.2-2.1ubuntu2.2_amd64.deb  404  Not Found [IP: 91.189.88.142 80]
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Aborting install.
==> ubuntu-focal: Running provisioner: install (shell)...
    ubuntu-focal: Running: inline script
    ubuntu-focal: .____    ._____.               ___________.__                
    ubuntu-focal: |    |   |__\_ |_________   ___\__    ___/|__| _____   ____  
    ubuntu-focal: |    |   |  || __ \_  __ \_/ __ \|    |   |  |/     \_/ __ \ 
    ubuntu-focal: |    |___|  || \_\ \  | \/\  ___/|    |   |  |  Y Y  \  ___/ 
    ubuntu-focal: |_______ \__||___  /__|    \___  >____|   |__|__|_|  /\___  >
    ubuntu-focal:         \/       \/            \/                  \/     \/
    ubuntu-focal: Detecting distribution and release ...
    ubuntu-focal: Detected distribution id: ubuntu
    ubuntu-focal: Detected distribution release id: 20.04
    ubuntu-focal: Detected distribution description: Ubuntu 20.04.1 LTS
    ubuntu-focal: Validating dist-code: ubuntu-20.04
    ubuntu-focal: Using distribution id "ubuntu", release code "focal"
    ubuntu-focal: Detecting init system type ...
    ubuntu-focal: Detected path to PID=1 process: /usr/lib/systemd/systemd
    ubuntu-focal: Detected package name for PID=1 process: unknown
    ubuntu-focal: Detected init system type: systemd
    ubuntu-focal: -----------------------------------------------------
    ubuntu-focal:          * Installing External Dependencies *        
    ubuntu-focal: -----------------------------------------------------
    ubuntu-focal: add-apt-repository -y ppa:libretime/libretime
    ubuntu-focal: Hit:1 http://archive.ubuntu.com/ubuntu focal InRelease
    ubuntu-focal: Ign:2 http://ppa.launchpad.net/libretime/libretime/ubuntu focal InRelease
    ubuntu-focal: Get:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB]
    ubuntu-focal: Get:4 http://security.ubuntu.com/ubuntu focal-security InRelease [109 kB]
    ubuntu-focal: Err:5 http://ppa.launchpad.net/libretime/libretime/ubuntu focal Release
    ubuntu-focal:   404  Not Found [IP: 91.189.95.85 80]
    ubuntu-focal: Get:6 http://archive.ubuntu.com/ubuntu focal-backports InRelease [101 kB]
    ubuntu-focal: Get:7 http://security.ubuntu.com/ubuntu focal-security/main i386 Packages [183 kB]
    ubuntu-focal: Get:8 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages [807 kB]
    ubuntu-focal: Get:9 http://security.ubuntu.com/ubuntu focal-security/main amd64 Packages [489 kB]
    ubuntu-focal: Get:10 http://archive.ubuntu.com/ubuntu focal-updates/main i386 Packages [421 kB]
    ubuntu-focal: Get:11 http://archive.ubuntu.com/ubuntu focal-updates/main Translation-en [195 kB]
    ubuntu-focal: Get:12 http://security.ubuntu.com/ubuntu focal-security/main Translation-en [107 kB]
    ubuntu-focal: Get:13 http://archive.ubuntu.com/ubuntu focal-updates/restricted amd64 Packages [146 kB]
    ubuntu-focal: Get:14 http://security.ubuntu.com/ubuntu focal-security/restricted amd64 Packages [123 kB]
    ubuntu-focal: Get:15 http://archive.ubuntu.com/ubuntu focal-updates/restricted i386 Packages [16.2 kB]
    ubuntu-focal: Get:16 http://archive.ubuntu.com/ubuntu focal-updates/restricted Translation-en [21.9 kB]
    ubuntu-focal: Get:17 http://archive.ubuntu.com/ubuntu focal-updates/universe i386 Packages [549 kB]
    ubuntu-focal: Get:18 http://security.ubuntu.com/ubuntu focal-security/restricted i386 Packages [14.9 kB]
    ubuntu-focal: Get:19 http://security.ubuntu.com/ubuntu focal-security/restricted Translation-en [18.0 kB]
    ubuntu-focal: Get:20 http://security.ubuntu.com/ubuntu focal-security/universe i386 Packages [422 kB]
    ubuntu-focal: Get:21 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 Packages [740 kB]
    ubuntu-focal: Get:22 http://security.ubuntu.com/ubuntu focal-security/universe amd64 Packages [536 kB]
    ubuntu-focal: Get:23 http://archive.ubuntu.com/ubuntu focal-updates/universe Translation-en [153 kB]
    ubuntu-focal: Get:24 http://archive.ubuntu.com/ubuntu focal-updates/multiverse i386 Packages [4,180 B]
    ubuntu-focal: Get:25 http://archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 Packages [16.9 kB]
    ubuntu-focal: Get:26 http://archive.ubuntu.com/ubuntu focal-updates/multiverse Translation-en [5,076 B]
    ubuntu-focal: Get:27 http://archive.ubuntu.com/ubuntu focal-backports/universe amd64 Packages [4,032 B]
    ubuntu-focal: Get:28 http://archive.ubuntu.com/ubuntu focal-backports/universe i386 Packages [2,932 B]
    ubuntu-focal: Get:29 http://security.ubuntu.com/ubuntu focal-security/universe Translation-en [76.1 kB]
    ubuntu-focal: Get:30 http://security.ubuntu.com/ubuntu focal-security/multiverse i386 Packages [2,336 B]
    ubuntu-focal: Get:31 http://security.ubuntu.com/ubuntu focal-security/multiverse amd64 Packages [10.4 kB]
    ubuntu-focal: Get:32 http://security.ubuntu.com/ubuntu focal-security/multiverse Translation-en [2,876 B]
    ubuntu-focal: Reading package lists...
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: The repository 'http://ppa.launchpad.net/libretime/libretime/ubuntu focal Release' does not have a Release file.
    ubuntu-focal:  * Reading requirements-ubuntu-focal.apt...
    ubuntu-focal: apt-get -q update
    ubuntu-focal: Hit:1 http://security.ubuntu.com/ubuntu focal-security InRelease
    ubuntu-focal: Ign:2 http://ppa.launchpad.net/libretime/libretime/ubuntu focal InRelease
    ubuntu-focal: Hit:3 http://archive.ubuntu.com/ubuntu focal InRelease
    ubuntu-focal: Hit:4 http://archive.ubuntu.com/ubuntu focal-updates InRelease
    ubuntu-focal: Err:5 http://ppa.launchpad.net/libretime/libretime/ubuntu focal Release
    ubuntu-focal:   404  Not Found [IP: 91.189.95.85 80]
    ubuntu-focal: Hit:6 http://archive.ubuntu.com/ubuntu focal-backports InRelease
    ubuntu-focal: Reading package lists...
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: The repository 'http://ppa.launchpad.net/libretime/libretime/ubuntu focal Release' does not have a Release file.
    ubuntu-focal: Detected apt-get version as: 2.0.2ubuntu0.2
    ubuntu-focal: Using apt-get force options: --allow-downgrades --allow-remove-essential --allow-change-held-packages
    ubuntu-focal: DEBIAN_FRONTEND=noninteractive apt-get -y -m --allow-downgrades --allow-remove-essential --allow-change-held-packages install apache2 build-essential coreutils curl ecasound flac gstreamer1.0-plugins-bad gstreamer1.0-plugins-good gstreamer1.0-plugins-ugly icecast2 lame libao-ocaml libapache2-mod-php7.3 libcamomile-ocaml-data libfaad2 libffi-dev libmad-ocaml libopus0 libportaudio2 libpulse0 libsamplerate0 libsoundtouch-ocaml libssl-dev libtaglib-ocaml liquidsoap lsb-release lsof mpg123 patch php7.3 php7.3-curl php7.3-gd php7.3-pgsql php-apcu php-bcmath php-mbstring php-pear postgresql postgresql-client pwgen python3 python3-dev python3-gst-1.0 python3-pika python3-pip python3-virtualenv python3-cairo rabbitmq-server silan sysvinit-utils unzip vorbisgain vorbis-tools vorbis-tools xmlstarlet zip 
    ubuntu-focal: Reading package lists...
    ubuntu-focal: Building dependency tree...
    ubuntu-focal: Reading state information...
    ubuntu-focal: Package libapache2-mod-php7.3 is not available, but is referred to by another package.
    ubuntu-focal: This may mean that the package is missing, has been obsoleted, or
    ubuntu-focal: is only available from another source
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Package 'libapache2-mod-php7.3' has no installation candidate
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Unable to locate package php7.3-curl
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Couldn't find any package by glob 'php7.3-curl'
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Couldn't find any package by regex 'php7.3-curl'
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Unable to locate package php7.3-gd
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Couldn't find any package by glob 'php7.3-gd'
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Couldn't find any package by regex 'php7.3-gd'
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Unable to locate package php7.3-pgsql
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Couldn't find any package by glob 'php7.3-pgsql'
    ubuntu-focal: E
    ubuntu-focal: : 
    ubuntu-focal: Couldn't find any package by regex 'php7.3-pgsql'
    ubuntu-focal: DEBIAN_FRONTEND=noninteractive apt-get -y -m install git
    ubuntu-focal: Reading package lists...
    ubuntu-focal: Building dependency tree...
    ubuntu-focal: Reading state information...
    ubuntu-focal: git is already the newest version (1:2.25.1-1ubuntu3).
    ubuntu-focal: 0 upgraded, 0 newly installed, 0 to remove and 64 not upgraded.
    ubuntu-focal: ./install: line 779: php: command not found
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

**Comment by Robbt** [comment]
So I figured out the first issue - need to make this change
`
 
-    if [ "$is_ubuntu_focal" = "t" -a ! -x /usr/bin/php7.3 ]; then
+    if [ "$is_ubuntu_focal" = "true" -a ! -x /usr/bin/php7.3 ]; then
         loud "Adding PHP PPA for PHP 7.3"
-       loudCmd "add-apt-repository -y ppa:ppa:ondre/php"
+       loudCmd "add-apt-repository -y ppa:ondrej/php"
     fi
`

and also 
`
diff --git a/installer/lib/requirements-ubuntu-focal.apt b/installer/lib/requirements-ubuntu-focal.apt
index 901a7eb15..3acce27fb 100644
--- a/installer/lib/requirements-ubuntu-focal.apt
+++ b/installer/lib/requirements-ubuntu-focal.apt
@@ -31,9 +31,9 @@ php7.3
 php7.3-curl
 php7.3-gd
 php7.3-pgsql
-php-apcu
-php-bcmath
-php-mbstring
+php7.3-apcu
+php7.3-bcmath
+php7.3-mbstring
`

but then it also needs to have #1110 merged because I ran into these issues.

**Comment by Robbt** [comment]
Ok I pushed my install scripts but I ran into issues with Liquidsoap 1.4 - see #1138 - so it is still not working.

**Comment by GeertSurkijn** [comment]
FYI, 
2021-04-01 02_49_08-gvrlibretime westeurope cloudapp azure com__config
I also got stuck on Liquidsoap 1.4. All installation processes passed, The check at the end stated Liquidsoap is not running. I cannot start it. Looks like all changes are there, the version of Liquidsoap which is being installed seems to be 1.4.1. The log of Liquidsoap itself is not created nor logging.

**Comment by paddatrapper** [comment]
I can't replicate `libretime-liquidsoap` failing to start, but I do get liquidsoap not running under that process. The `airttime-liquidsoap` python script runs, but nothing seems to be able to connect to liquidsoap

**Comment by paddatrapper** [comment]
Running into

```
Apr 01 08:31:18 vagrant airtime-liquidsoap[54231]: Warning: illegal backslash escape in string.
Apr 01 08:31:18 vagrant airtime-liquidsoap[54231]: At /usr/local/lib/python3.8/dist-packages/airtime_playout-1.0-py3.8.egg/liquidsoap/1.4/ls_lib.liq, line 30, char 5-20:
Apr 01 08:31:18 vagrant airtime-liquidsoap[54231]: Error 4: Undefined variable list.mem_assoc
```

@hairmare any idea what the issue is?

**Comment by GeertSurkijn** [comment]
In that file ls_lib.liq on line 30, is this a typo? I am not familar with ocaml, but list.mem_assoc (can it be it must be List.mem_assoc ???)
Capital L?

`  if list.mem_assoc("mapped", m) then
    #protection against applying this function twice. It shouldn't be happening
    #and bug file with Liquidsoap.
    m
  else
`

**Comment by paddatrapper** [comment]
Seems it is valid liquidsoap - https://www.liquidsoap.info/doc-1.4.4/reference.html#list.mem_assoc

**Comment by hashanchamikara** [comment]
> FYI,
> 2021-04-01 02_49_08-gvrlibretime westeurope cloudapp azure com__config
> I also got stuck on Liquidsoap 1.4. All installation processes passed, The check at the end stated Liquidsoap is not running. I cannot start it. Looks like all changes are there, the version of Liquidsoap which is being installed seems to be 1.4.1. The log of Liquidsoap itself is not created nor logging.

I got same problem

**Comment by paddatrapper** [comment]
Full liquidsoap error:

```
Jun 27 15:30:44 vagrant airtime-liquidsoap[3081]: Warning: illegal backslash escape in string.
Jun 27 15:30:44 vagrant airtime-liquidsoap[3081]: At /usr/local/lib/python3.8/dist-packages/airtime_playout-1.0-py3.8.egg/liquidsoap/1.4/ls_lib.liq, line 30, char 5-20:
Jun 27 15:30:44 vagrant airtime-liquidsoap[3081]: Error 4: Undefined variable list.mem_assoc
```

**Comment by paddatrapper** [comment]
The liquidsoap failure is due to liquidsoap being broken in 20.04. I'm busy preparing a fix so that the same issue does not occur in Debian Bullseye and will look at either getting it updated in Ubuntu or shipping liquidsoap in our PPA. Issue details: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=901886

**Comment by jooola** [comment]
@paddatrapper Could you rebase this PR ?

Also do we have an idea about this error with Silan ?

We can't ask Debian to ship liquidsoap 1.4.4 right ? Even through patches ?

**Comment by paddatrapper** [comment]
> @paddatrapper Could you rebase this PR ?

Will do. Just busy trying to get the liquidsoap fix into Ubuntu 20.04 at the moment.

> We can't ask Debian to ship liquidsoap 1.4.4 right ? Even through patches ? Saw you recent work, you uploaded it to unstable !

It will be in Bookworm if there are no later liquidsoap releases. Though liquidsoap version 2 will be out shortly, so that is most likely going to be the version in Bookworm. We'll have to spend some time getting things working with v2...

**Comment by jooola** [comment]
> It will be in Bookworm if there are no later liquidsoap releases. Though liquidsoap version 2 will be out shortly, so that is most likely going to be the version in Bookworm. We'll have to spend some time getting things working with v2...

I'll be happy to make the liquidsoap v2 migration.

**Comment by paddatrapper** [comment]
Rebased

File: Vagrantfile [hunk 1]
```diff
@@ -48,6 +48,11 @@ Vagrant.configure("2") do |config|
     os.vm.box = "bento/ubuntu-18.04"
     provision_libretime(os, "debian.sh", installer_args)
   end
+  # TODO: Fix the script to install PHP 7.3 from Ondrej PPA
```
Comment (jooola) [inline Vagrantfile:51]: Some leftover.

File: install [hunk 2]
```diff
@@ -1063,10 +1073,12 @@ else
 fi
 
 # Enable Apache modules
-if $is_debian_buster; then
+elif $is_debian_buster || $is_ubuntu_focal; then
```
Comment (jooola) [inline install:1076]: Is that `elif` really intended ?

File: install [hunk 3]
```diff
@@ -1063,10 +1073,12 @@ else
...
     loudCmd "a2enmod rewrite php7.3 proxy proxy_http"
 elif $is_ubuntu_bionic; then
     loudCmd "a2enmod rewrite php7.2 proxy proxy_http"
+elif $is_ubuntu_xenial; then
```
Comment (jooola) [inline install:1080]: Was dropped.

**Review by jooola — CHANGES_REQUESTED** [review]

**Comment by paddatrapper** [comment]
Verification step has been completed for liquidsoap fix getting into Focal. Hopefully that should be uploaded into stable-updates in the next couple days. After that, this should work and run on Ubuntu 20.04

**Comment by paddatrapper** [comment]
Liquidsoap in Focal is now fixed. Once the Focal docker image builds, I can get the CI passing

**Comment by paddatrapper** [comment]
Adding a show fails with 

```
2021-09-12T18:22:14+00:00 ERR (3): localhost [Logging.php:166 - loggingShutdownCallback()] - Uncaught Error: Class 'DOMDocument' not found in /vagrant/vendor/zf1s/zend-xml/library/Zend/Xml/Security.php:82
Stack trace:
#0 /vagrant/vendor/zf1s/zend-xml/library/Zend/Xml/Security.php(150): Zend_Xml_Security::scan('<?xml version="...', NULL)
#1 /vagrant/vendor/zf1s/zend-locale/library/Zend/Locale/Data.php(160): Zend_Xml_Security::scanFile('/vagrant/vendor...')
#2 /vagrant/vendor/zf1s/zend-locale/library/Zend/Locale/Data.php(232): Zend_Locale_Data::_findRoute('en_ZA', '/ldml/dates/cal...', 'type', false, Array)
#3 /vagrant/vendor/zf1s/zend-locale/library/Zend/Locale/Data.php(491): Zend_Locale_Data::_getFile('en_ZA', '/ldml/dates/cal...', 'type')
#4 /vagrant/vendor/zf1s/zend-locale/library/Zend/Locale/Format.php(852): Zend_Locale_Data::getList('en_ZA', 'day')
#5 /vagrant/vendor/zf1s/zend-locale/library/Zend/Locale/Format.php(1162): Zend_Locale_Format::_parseDate('2021-09-12', Array)
#6 /vagrant/vendor/zf1s/zend-date/library/Zend/Date.php(4787): Zend_Locale_Format::getDate('2021-09-12', Array)
#7 /vagrant/ve at /vagrant/vendor/zf1s/zend-xml/library/Zend/Xml/Security.php:82
array (
  'type' => 1,
  'message' => 'Uncaught Error: Class \'DOMDocument\' not found in /vagrant/vendor/zf1s/zend-xml/library/Zend/Xml/Security.php:82
Stack trace:
#0 /vagrant/vendor/zf1s/zend-xml/library/Zend/Xml/Security.php(150): Zend_Xml_Security::scan(\'<?xml version="...\', NULL)
#1 /vagrant/vendor/zf1s/zend-locale/library/Zend/Locale/Data.php(160): Zend_Xml_Security::scanFile(\'/vagrant/vendor...\')
#2 /vagrant/vendor/zf1s/zend-locale/library/Zend/Locale/Data.php(232): Zend_Locale_Data::_findRoute(\'en_ZA\', \'/ldml/dates/cal...\', \'type\', false, Array)
#3 /vagrant/vendor/zf1s/zend-locale/library/Zend/Locale/Data.php(491): Zend_Locale_Data::_getFile(\'en_ZA\', \'/ldml/dates/cal...\', \'type\')
#4 /vagrant/vendor/zf1s/zend-locale/library/Zend/Locale/Format.php(852): Zend_Locale_Data::getList(\'en_ZA\', \'day\')
#5 /vagrant/vendor/zf1s/zend-locale/library/Zend/Locale/Format.php(1162): Zend_Locale_Format::_parseDate(\'2021-09-12\', Array)
#6 /vagrant/vendor/zf1s/zend-date/library/Zend/Date.php(4787): Zend_Locale_Format::getDate(\'2021-09-12\', Array)
#7 /vagrant/ve',
  'file' => '/vagrant/vendor/zf1s/zend-xml/library/Zend/Xml/Security.php',
  'line' => 82,
)
```

**Comment by paddatrapper** [comment]
Still at a loss as to why the streaming is so choppy... I don't know the liquidsoap side of things enough to debug what is going wrong

**Comment by paddatrapper** [comment]
@jooola looks like it is liquidsoap 1.4.1... It worked with 1.4.4. Hopefully the versions of dependencies in Focal work, so I can upload the 1.4.4 version to the PPA

File: legacy/packages.ini [hunk 4]
```diff
@@ -38,3 +38,19 @@ php7.3-gd = buster
...
+php7.4-apcu-bc = focal
+php7.4-bcmath = focal
+php7.4-cli = focal
+php7.4-common = focal
+php7.4-curl = focal
+php7.4-dev = focal
+php7.4-gd = focal
+php7.4-mbstring = focal
+php7.4-pgsql = focal
+php7.4-xml = buster, focal
```
Comment (jooola) [inline legacy/packages.ini:56]: I think we missed the extra `buster,` here right ?

Comment (paddatrapper) [reply legacy/packages.ini:56]: Yup, seems so

**Pull request merged**